### ISSUE TITLE
Multiple dependency conflicts in quick-media, which may lead to the unexpected semantic behaviors

### ISSUE THREAD

**Issue by HelloCoCooo** [liuyueyi/quick-media#41]
Hi, in **_quick-media-0.008\plugins\phantom-plugin_** there are multiple versions of library **_org.seleniumhq.selenium:selenium-remote-driver:_**, i.e., **_org.seleniumhq.selenium:selenium-remote-driver:jar:2.53.1_** and **_org.seleniumhq.selenium:selenium-remote-driver:jar:3.1.0_**.  According to Maven management strategy, only **_org.seleniumhq.selenium:selenium-remote-driver:jar:2.53.1_** can be loaded.

By static analysis, library **_org.seleniumhq.selenium:selenium-java:jar:2.53.1:compile_** expects to reference the methods defined in class **_org.openqa.selenium.remote.DesiredCapabilities_**  included in version **_org.seleniumhq.selenium:selenium-remote-driver:jar:2.53.1_** via the following invocation path:

```
<com.github.hui.quick.plugin.phantom.Html2ImageByJsWrapper: void <clinit>()> D:\testcase\TestProject\quick-media-0.008\plugins\phantom-plugin\target\classes
<com.github.hui.quick.plugin.phantom.Html2ImageByJsWrapper: org.openqa.selenium.phantomjs.PhantomJSDriver getPhantomJs()> D:\testcase\TestProject\quick-media-0.008\plugins\phantom-plugin\target\classes
<org.openqa.selenium.remote.DesiredCapabilities: void setCapability(java.lang.String,java.lang.String)> D:\cEnvironment\repository\org\seleniumhq\selenium\selenium-api\2.53.1\selenium-api-2.53.1.jar
<org.apache.commons.collections.map.FixedSizeSortedMap: java.lang.Object put(java.lang.Object,java.lang.Object)> D:\cEnvironment\repository\commons-collections\commons-collections\3.2.2\commons-collections-3.2.2.jar
<net.sourceforge.htmlunit.corejs.javascript.NativeObject: boolean containsKey(java.lang.Object)> D:\cEnvironment\repository\net\sourceforge\htmlunit\htmlunit-core-js\2.17\htmlunit-core-js-2.17.jar
<net.sourceforge.htmlunit.corejs.javascript.IdScriptableObject: boolean has(java.lang.String,net.sourceforge.htmlunit.corejs.javascript.Scriptable)> D:\cEnvironment\repository\net\sourceforge\htmlunit\htmlunit-core-js\2.17\htmlunit-core-js-2.17.jar
<net.sourceforge.htmlunit.corejs.javascript.regexp.NativeRegExpCtor: java.lang.Object getInstanceIdValue(int)> D:\cEnvironment\repository\net\sourceforge\htmlunit\htmlunit-core-js\2.17\htmlunit-core-js-2.17.jar
<org.openqa.selenium.remote.DesiredCapabilities: java.lang.String toString()>

```
 However, **_org.seleniumhq.selenium:selenium-remote-driver:jar:2.53.1_** would not be loaded due to dependency conflicts, and the actual loaded version **_org.seleniumhq.selenium:selenium-remote-driver:jar:3.1.0_** does not contain this class at all!!
To my surprise, there is no classnotfoundexeceptions here!!
What a Coincidence, another Jar file **_org.seleniumhq.selenium:selenium-api:2.53.1_** also contain a class with the same fully qualified name "**_org.openqa.selenium.remote.DesiredCapabilities_**".
As a result, it is forced referenced by library **_org.seleniumhq.selenium:selenium-java:jar:2.53.1:compile_**  via the above invocation path. 
I think it is a danger signal for project maintenance.

By further analyzing, the expected callee _**org.openqa.selenium.remote.DesiredCapabilities: java.lang.String toString()**_ in shadowed version **_org.seleniumhq.selenium:selenium-remote-driver:jar:2.53.1_** , has different implementations from the actual callee with the same signature (same method names, same paremeters) included in the unexpected (but actual loaded) version **_org.seleniumhq.selenium:selenium-api:2.53.1_**, which leads to different behaviors.

### **Solution:**
Upgrade  **_org.seleniumhq.selenium:selenium-remote-driver:jar:2.53.1_** to **_3.1.0_**, to keep the version consistency.

### Dependency tree---

[INFO] --- maven-dependency-plugin:2.10:tree (default-cli) @ phantom-plugin ---
[WARNING] Using Maven 2 dependency tree to get verbose output, which may be inconsistent with actual Maven 3 resolution
[INFO] com.github.hui.media:phantom-plugin:jar:1.0
[INFO] +- **_org.seleniumhq.selenium:selenium-java:jar:2.53.1:compile_**
[INFO] |  +- org.seleniumhq.selenium:selenium-chrome-driver:jar:2.53.1:compile
[INFO] |  |  \- (_**org.seleniumhq.selenium:selenium-remote-driver:jar:2.53.1:compile**_ - **_version managed from 3.1.0;_** omitted for duplicate)
[INFO] |  +- org.seleniumhq.selenium:selenium-edge-driver:jar:2.53.1:compile
[INFO] |  |  +- (**org.seleniumhq.selenium:selenium-remote-driver:jar:2.53.1:compile** - omitted for duplicate)
[INFO] |  |  +- commons-io:commons-io:jar:2.4:compile
[INFO] |  |  \- org.apache.commons:commons-exec:jar:1.3:compile
[INFO] |  +- org.seleniumhq.selenium:htmlunit-driver:jar:2.21:compile
[INFO] |  |  +- net.sourceforge.htmlunit:htmlunit:jar:2.21:compile
[INFO] |  |  |  +- xalan:xalan:jar:2.7.2:compile
[INFO] |  |  |  |  \- xalan:serializer:jar:2.7.2:compile
[INFO] |  |  |  +- org.apache.commons:commons-lang3:jar:3.5:compile (version managed from 3.4)
[INFO] |  |  |  +- (org.apache.httpcomponents:httpclient:jar:4.5.3:compile - version managed from 4.5.2; omitted for duplicate)
[INFO] |  |  |  +- org.apache.httpcomponents:httpmime:jar:4.5.3:compile (version managed from 4.5.2)
[INFO] |  |  |  |  \- (org.apache.httpcomponents:httpclient:jar:4.5.3:compile - version managed from 4.5.2; omitted for duplicate)
[INFO] |  |  |  +- commons-codec:commons-codec:jar:1.10:compile
[INFO] |  |  |  +- net.sourceforge.htmlunit:htmlunit-core-js:jar:2.17:compile
[INFO] |  |  |  +- net.sourceforge.htmlunit:neko-htmlunit:jar:2.21:compile
[INFO] |  |  |  |  \- xerces:xercesImpl:jar:2.11.0:compile
[INFO] |  |  |  |     \- xml-apis:xml-apis:jar:1.4.01:compile
[INFO] |  |  |  +- net.sourceforge.cssparser:cssparser:jar:0.9.18:compile
[INFO] |  |  |  |  \- org.w3c.css:sac:jar:1.3:compile
[INFO] |  |  |  +- (commons-io:commons-io:jar:2.4:compile - omitted for duplicate)
[INFO] |  |  |  \- org.eclipse.jetty.websocket:websocket-client:jar:9.4.5.v20170502:compile (version managed from 9.2.15.v20160210)
[INFO] |  |  |     +- org.eclipse.jetty:jetty-util:jar:9.4.5.v20170502:compile
[INFO] |  |  |     +- org.eclipse.jetty:jetty-io:jar:9.4.5.v20170502:compile
[INFO] |  |  |     |  \- (org.eclipse.jetty:jetty-util:jar:9.4.5.v20170502:compile - omitted for duplicate)
[INFO] |  |  |     +- org.eclipse.jetty:jetty-client:jar:9.4.5.v20170502:compile
[INFO] |  |  |     |  +- org.eclipse.jetty:jetty-http:jar:9.4.5.v20170502:compile
[INFO] |  |  |     |  |  +- (org.eclipse.jetty:jetty-util:jar:9.4.5.v20170502:compile - omitted for duplicate)
[INFO] |  |  |     |  |  \- (org.eclipse.jetty:jetty-io:jar:9.4.5.v20170502:compile - omitted for duplicate)
[INFO] |  |  |     |  \- (org.eclipse.jetty:jetty-io:jar:9.4.5.v20170502:compile - omitted for duplicate)
[INFO] |  |  |     \- org.eclipse.jetty.websocket:websocket-common:jar:9.4.5.v20170502:compile
[INFO] |  |  |        +- org.eclipse.jetty.websocket:websocket-api:jar:9.4.5.v20170502:compile
[INFO] |  |  |        +- (org.eclipse.jetty:jetty-util:jar:9.4.5.v20170502:compile - omitted for duplicate)
[INFO] |  |  |        \- (org.eclipse.jetty:jetty-io:jar:9.4.5.v20170502:compile - omitted for duplicate)
[INFO] |  |  \- commons-collections:commons-collections:jar:3.2.2:compile
[INFO] |  +- org.seleniumhq.selenium:selenium-firefox-driver:jar:2.53.1:compile
[INFO] |  |  +- (_**org.seleniumhq.selenium:selenium-remote-driver:jar:2.53.1:compile**_ - omitted for duplicate)
[INFO] |  |  +- (commons-io:commons-io:jar:2.4:compile - omitted for duplicate)
[INFO] |  |  \- (org.apache.commons:commons-exec:jar:1.3:compile - omitted for duplicate)
[INFO] |  +- org.seleniumhq.selenium:selenium-ie-driver:jar:2.53.1:compile
[INFO] |  |  +- net.java.dev.jna:jna:jar:4.2.2:compile
[INFO] |  |  +- net.java.dev.jna:jna-platform:jar:4.1.0:compile
[INFO] |  |  |  \- (net.java.dev.jna:jna:jar:4.2.2:compile - version managed from 4.1.0; omitted for duplicate)
[INFO] |  |  \- (_**org.seleniumhq.selenium:selenium-remote-driver:jar:2.53.1:compile**_ - omitted for duplicate)
[INFO] |  +- org.seleniumhq.selenium:selenium-safari-driver:jar:2.53.1:compile
[INFO] |  |  +- (_**org.seleniumhq.selenium:selenium-remote-driver:jar:2.53.1:compile**_ - omitted for duplicate)
[INFO] |  |  \- io.netty:netty:jar:3.5.7.Final:compile
[INFO] |  +- org.seleniumhq.selenium:selenium-support:jar:2.53.1:compile
[INFO] |  |  \- (**org.seleniumhq.selenium:selenium-remote-driver:jar:2.53.1:compile** - omitted for duplicate)
[INFO] |  \- org.seleniumhq.selenium:selenium-leg-rc:jar:2.53.1:compile
[INFO] |     \- (_**org.seleniumhq.selenium:selenium-remote-driver:jar:2.53.1:compile**_ - omitted for duplicate)
[INFO] +- com.github.detro:ghostdriver:jar:2.1.0:compile
[INFO] |  \- _**org.seleniumhq.selenium:selenium-remote-driver:jar:2.53.1:compile**_
[INFO] |     +- cglib:cglib-nodep:jar:2.1_3:compile
[INFO] |     +- com.google.code.gson:gson:jar:2.8.0:compile (version managed from 2.3.1)
[INFO] |     +- _**org.seleniumhq.selenium:selenium-api:jar:2.53.1:compile**_
[INFO] |     |  +- (com.google.guava:guava:jar:19.0:compile - omitted for duplicate)
[INFO] |     |  \- (com.google.code.gson:gson:jar:2.8.0:compile - version managed from 2.3.1; omitted for duplicate)
[INFO] |     +- org.apache.httpcomponents:httpclient:jar:4.5.3:compile (version managed from 4.5.1)
[INFO] |     |  +- org.apache.httpcomponents:httpcore:jar:4.4.6:compile
[INFO] |     |  \- (commons-codec:commons-codec:jar:1.10:compile - version managed from 1.9; omitted for duplicate)
[INFO] |     +- (com.google.guava:guava:jar:19.0:compile - omitted for conflict with 21.0)
[INFO] |     +- (org.apache.commons:commons-exec:jar:1.3:compile - omitted for duplicate)
[INFO] |     +- (net.java.dev.jna:jna:jar:4.2.2:compile - version managed from 4.1.0; omitted for duplicate)
[INFO] |     \- (net.java.dev.jna:jna-platform:jar:4.1.0:compile - omitted for duplicate)
[INFO] +- com.google.guava:guava:jar:21.0:compile
[INFO] +- org.slf4j:slf4j-api:jar:1.7.25:compile
[INFO] +- commons-logging:commons-logging:jar:1.2:compile
[INFO] +- ch.qos.logback:logback-classic:jar:1.1.7:compile
[INFO] |  +- ch.qos.logback:logback-core:jar:1.1.11:compile (version managed from 1.1.7)
[INFO] |  \- (org.slf4j:slf4j-api:jar:1.7.25:compile - version managed from 1.7.20; omitted for duplicate)
[INFO] \- org.springframework.boot:spring-boot-starter-test:jar:1.5.4.RELEASE:test
[INFO]    +- org.springframework.boot:spring-boot-test:jar:1.5.4.RELEASE:test
[INFO]    |  \- org.springframework.boot:spring-boot:jar:1.5.4.RELEASE:test
[INFO]    |     +- (org.springframework:spring-core:jar:4.3.9.RELEASE:test - omitted for duplicate)
[INFO]    |     \- org.springframework:spring-context:jar:4.3.9.RELEASE:test
[INFO]    |        +- org.springframework:spring-aop:jar:4.3.9.RELEASE:test
[INFO]    |        |  +- (org.springframework:spring-beans:jar:4.3.9.RELEASE:test - omitted for duplicate)
[INFO]    |        |  \- (org.springframework:spring-core:jar:4.3.9.RELEASE:test - omitted for duplicate)
[INFO]    |        +- org.springframework:spring-beans:jar:4.3.9.RELEASE:test
[INFO]    |        |  \- (org.springframework:spring-core:jar:4.3.9.RELEASE:test - omitted for duplicate)
[INFO]    |        +- (org.springframework:spring-core:jar:4.3.9.RELEASE:test - omitted for duplicate)
[INFO]    |        \- org.springframework:spring-expression:jar:4.3.9.RELEASE:test
[INFO]    |           \- (org.springframework:spring-core:jar:4.3.9.RELEASE:test - omitted for duplicate)
[INFO]    +- org.springframework.boot:spring-boot-test-autoconfigure:jar:1.5.4.RELEASE:test
[INFO]    |  +- (org.springframework.boot:spring-boot-test:jar:1.5.4.RELEASE:test - omitted for duplicate)
[INFO]    |  \- org.springframework.boot:spring-boot-autoconfigure:jar:1.5.4.RELEASE:test
[INFO]    |     \- (org.springframework.boot:spring-boot:jar:1.5.4.RELEASE:test - omitted for duplicate)
[INFO]    +- com.jayway.jsonpath:json-path:jar:2.2.0:test
[INFO]    |  +- net.minidev:json-smart:jar:2.2.1:test
[INFO]    |  |  \- net.minidev:accessors-smart:jar:1.1:test
[INFO]    |  |     \- org.ow2.asm:asm:jar:5.0.3:test
[INFO]    |  \- (org.slf4j:slf4j-api:jar:1.7.25:test - version managed from 1.7.16; omitted for duplicate)
[INFO]    +- junit:junit:jar:4.12:test
[INFO]    |  \- (org.hamcrest:hamcrest-core:jar:1.3:test - version managed from 1.1; omitted for duplicate)
[INFO]    +- org.assertj:assertj-core:jar:2.6.0:test
[INFO]    +- org.mockito:mockito-core:jar:1.10.19:test
[INFO]    |  +- (org.hamcrest:hamcrest-core:jar:1.3:test - version managed from 1.1; omitted for duplicate)
[INFO]    |  \- org.objenesis:objenesis:jar:2.1:test
[INFO]    +- org.hamcrest:hamcrest-core:jar:1.3:test
[INFO]    +- org.hamcrest:hamcrest-library:jar:1.3:test
[INFO]    |  \- (org.hamcrest:hamcrest-core:jar:1.3:test - version managed from 1.1; omitted for duplicate)
[INFO]    +- org.skyscreamer:jsonassert:jar:1.4.0:test
[INFO]    |  \- com.vaadin.external.google:android-json:jar:0.0.20131108.vaadin1:test
[INFO]    +- org.springframework:spring-core:jar:4.3.9.RELEASE:test
[INFO]    \- org.springframework:spring-test:jar:4.3.9.RELEASE:test
[INFO]       \- (org.springframework:spring-core:jar:4.3.9.RELEASE:test - omitted for duplicate)

Best regards,
Coco

**Comment by HelloCoCooo** [comment]
May I pull a request to fix this issue?

**Comment by liuyueyi** [comment]
the plugins's parent  inherit `spring-boot-starter-parent`,  i already fix it, and upgrade plugin version to 1.1

here is dependencies of phantom-plugin

![image](https://user-images.githubusercontent.com/5125892/56401042-96b50c80-6289-11e9-8b68-638690eef37f.png)

**Comment by HelloCoCooo** [comment]
@liuyueyi Thank you for your fixing.
